presto-hive/src/main/java/com/facebook/presto/hive/ManifestPartitionLoader.java

```diff
@@ -0,0 +1,230 @@
+                fileSystem,
+                inputFormat,
+                pathDomain,
+                getNodeSelectionStrategy(session),
+                getMaxInitialSplitSize(session),
+                false,
+                new HiveSplitPartitionInfo(
+                        storage,
+                        path.toUri(),
+                        partitionKeys,
+                        partitionName,
+                        partitionDataColumnCount,
+                        partition.getTableToPartitionMapping(),
+                        Optional.empty(),
+                        partition.getRedundantColumnDomains()),
+                schedulerUsesHostAddresses,
+                partition.getEncryptionInformation());
+    }
+
+    private void validateManifest(HivePartitionMetadata partition, Path path, List<String> manifestFileNames, List<Long> manifestFileSizes)
+            throws IOException
+    {
+        ExtendedFileSystem fileSystem = hdfsEnvironment.getFileSystem(hdfsContext, path);
+        HiveDirectoryContext hiveDirectoryContext = new HiveDirectoryContext(recursiveDirWalkerEnabled ? RECURSE : IGNORED, false);
+
+        Iterator<HiveFileInfo> fileInfoIterator = directoryLister.list(fileSystem, table, path, namenodeStats, ignore -> true, hiveDirectoryContext);
+        int fileCount = 0;
+        while (fileInfoIterator.hasNext()) {
+            HiveFileInfo fileInfo = fileInfoIterator.next();
+            String fileName = fileInfo.getPath().getName();
+            if (!manifestFileNames.contains(fileName)) {
+                throw new PrestoException(
+                        MALFORMED_HIVE_FILE_STATISTICS,
+                        format("Filename = %s not stored in manifest. Partition = %s, TableName = %s",
+                                fileName,
+                                partition.getHivePartition().getPartitionId(),
+                                table.getTableName()));
+            }
+
+            int index = manifestFileNames.indexOf(fileName);
+            if (!manifestFileSizes.get(index).equals(fileInfo.getLength())) {
+                throw new PrestoException(
+                        MALFORMED_HIVE_FILE_STATISTICS,
+                        format("FilesizeFromManifest = %s is not equal to FilesizeFromStorage = %s. File = %s, Partition = %s, TableName = %s",
+                                manifestFileSizes.get(index),
+                                fileInfo.getLength(),
+                                fileName,
+                                partition.getHivePartition().getPartitionId(),
+                                table.getTableName()));
+            }
+
+            fileCount++;
+        }
+
+        if (fileCount != manifestFileNames.size()) {
+            throw new PrestoException(
+                    MALFORMED_HIVE_FILE_STATISTICS,
+                    format("Number of files in Manifest = %s is not equal to Number of files in storage = %s. Partition = %s, TableName = %s",
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

This code uses &#39;%s&#39; to format int: &#39;size&#39; expression, fileCount (defined at line 192). This is a potential locale-sensitive handling issue. It might cause errors in the handling and processing of the statement at line: 223. Consider formatting this data with &#39;%d&#39; instead.